src/fadt.rs

```diff
@@ -53,25 +53,53 @@ pub struct Fadt {
     month_alarm: u8,
     century: u8,
     iapc_boot_arch: u16,
-    reserved2: u8, // must be 0
+    _reserved2: u8, // must be 0
     flags: u32,
     reset_reg: GenericAddress,
     reset_value: u8,
     arm_boot_arch: u16,
     fadt_minor_version: u8,
-    x_firmware_control: u64,
-    x_dsdt_address: u64,
-    x_pm1a_event_block: GenericAddress,
-    x_pm1b_event_block: GenericAddress,
-    x_pm1a_control_block: GenericAddress,
-    x_pm1b_control_block: GenericAddress,
-    x_pm2_control_block: GenericAddress,
-    x_pm_timer_block: GenericAddress,
-    x_gpe0_block: GenericAddress,
-    x_gpe1_block: GenericAddress,
-    sleep_control_reg: GenericAddress,
-    sleep_status_reg: GenericAddress,
-    hypervisor_vendor_id: u64,
+    x_firmware_ctrl: ExtendedField<u64>,
+    x_dsdt_address: ExtendedField<u64>,
+    x_pm1a_event_block: ExtendedField<GenericAddress>,
+    x_pm1b_event_block: ExtendedField<GenericAddress>,
+    x_pm1a_control_block: ExtendedField<GenericAddress>,
+    x_pm1b_control_block: ExtendedField<GenericAddress>,
+    x_pm2_control_block: ExtendedField<GenericAddress>,
+    x_pm_timer_block: ExtendedField<GenericAddress>,
+    x_gpe0_block: ExtendedField<GenericAddress>,
+    x_gpe1_block: ExtendedField<GenericAddress>,
+    sleep_control_reg: ExtendedField<GenericAddress>,
+    sleep_status_reg: ExtendedField<GenericAddress>,
+    hypervisor_vendor_id: ExtendedField<u64>,
+}
+
+#[derive(Copy, Clone)]
+#[repr(C, packed)]
+union ExtendedField<T: Copy> {
+    field: T,
+    nothing: (),
+}
+
+impl<T: Copy> ExtendedField<T> {
+    fn new(field: T) -> Self {
+        ExtendedField { field }
+    }
+
+    /// Checks that the given revision is greater than `1` and then returns the field, otherwise
+    /// returning `None`.
+    ///
+    /// # Unsafety
+    ///
+    /// This is unsafe as the given `revision` could be incorrect or fabricated. However, it should
+    /// be safe if it represents the revision of the FADT.
```

Comment:
Not sure we want to mention the FADT in the docs for this - it's not specific to any one table